txnkv/transaction/txn.go

```diff
@@ -392,6 +402,138 @@ func (txn *KVTxn) IsPessimistic() bool {
 	return txn.isPessimistic
 }
 
+// IsPipelined returns true if it's a pipelined transaction.
+func (txn *KVTxn) IsPipelined() bool {
+	return txn.isPipelined
+}
+
+func (txn *KVTxn) SetPipelinedMemDB() error {
+	txn.isPipelined = true
+	txn.snapshot.SetPipelined()
+	// TODO: set the correct sessionID
+	committer, err := newTwoPhaseCommitter(txn, 0)
+	if err != nil {
+		return err
+	}
+	txn.committer = committer
+	// disable 1pc and async commit for pipelined txn.
+	txn.committer.setOnePC(false)
```

Comment:
WARN logs are needed if any incompatible features are enabled by user as they are problematic usage

Reply:
One pc and async commit fields are turned on in `committer.execute`.

Manually setting them to false is just a emphasis.

WARN logs only take effects when the `newTwoPhaseCommitter` returns committer with one-pc or async-commit on, which is impossible by now.

Reply:
Emm can we simply make assertions then?

Reply:
Changed to fatal log.